src/gosling-track/gosling-track.ts

```diff
@@ -327,6 +364,7 @@ function GoslingTrack(HGC: any, ...args: any[]): any {
                 this.gLegend.remove();
                 this.gLegend = null;
             }
+            this.mRangeBrush.remove();
```

Comment:
Can you leave a comment describing why removed twice if layout if circular?

Reply:
Left comment statements on the `remove()` and `rerender()` functions.

Usually, you only need to remove any graphics on the `remove()` function. This function is called when a track is removed from the GoslingComponent.

I am additionally removing the range brush on `rerender()`. This `rerender()` function is called when a spec for the current track is changed by a user. Since we do not support a circular brush at the moment, I am removing the linear brush when the spec is changed from linear to circular:

```ts
rerender(...) {
  if (this.options.spec.layout === 'circular') {
    // this means the spec has been changed to use circular layouts
    this.mRangeBrush.remove();
  }
```

For example, you can see that any range brushes that you made on a default example of the editor (i.e., "Visual Encoding") are being removed when you change the `layout` to `circular`.

Reply:
Ok! Thanks for the clarification.